src/utils/mod.rs

```diff
@@ -886,3 +913,66 @@ impl TryFrom<TpmtTkVerified> for TPMT_TK_VERIFIED {
         })
     }
 }
+
+/// Create the TPM2B_PUBLIC structure for a restricted decryption key.
+///
+/// * `symmetric` - Cipher to be used for decrypting children of the key
+/// * `key_bits` - Size in bits of the decryption key
+/// * `pub_exponent` - Public exponent of the RSA key. A value of 0 defaults to 2^16 + 1
+pub fn create_restricted_decryption_rsa_public(
+    symmetric: Cipher,
+    key_bits: u16,
+    pub_exponent: u32,
+) -> Result<TPM2B_PUBLIC> {
+    let rsa_parms = TpmsRsaParmsBuilder::new_restricted_decryption_key(
+        symmetric.into(),
+        key_bits,
+        pub_exponent,
+    )
+    .build()?;
+    let mut object_attributes = ObjectAttributes(0);
+    object_attributes.set_fixed_tpm(true);
+    object_attributes.set_fixed_parent(true);
+    object_attributes.set_sensitive_data_origin(true);
+    object_attributes.set_user_with_auth(true);
+    object_attributes.set_decrypt(true);
+    object_attributes.set_sign_encrypt(false);
+    object_attributes.set_restricted(true);
+
+    Tpm2BPublicBuilder::new()
+        .with_type(TPM2_ALG_RSA)
+        .with_name_alg(TPM2_ALG_SHA256)
```

Comment:
Should we indicate in the function documentation that those parameters are hardcoded or is it fine because it's a constant throughout the whole crate?

Reply:
It should be in the docs, sorry!